library/src/jvmMain/kotlin/com/badoo/reaktive/scheduler/FixedExecutorServiceStrategy.kt

```diff
@@ -0,0 +1,26 @@
+package com.badoo.reaktive.scheduler
+
+import java.util.concurrent.Executors
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.ThreadFactory
+import java.util.concurrent.atomic.AtomicInteger
+
+internal class FixedExecutorServiceStrategy(
+    threadCount: Int,
+    threadFactory: ThreadFactory
+) : ExecutorServiceStrategy {
+
+    private val pool =
+        List<Lazy<ScheduledExecutorService>>(threadCount) {
+            lazy { Executors.newSingleThreadScheduledExecutor(threadFactory) }
+        }
+
+    private val executorIndex = AtomicInteger()
+
+    override fun get(): ScheduledExecutorService =
+        pool[executorIndex.getAndIncrement() % pool.size].value
+
+    override fun recycle(executorService: ScheduledExecutorService) {
+        // no-op
```

Comment:
What do you think of calling shutdown? For FixedThreadPools the docs state that the threads will exist until shutdown is called. Also ExecutorService has shutdown called in the finalize method. I think it would be a good idea to more reliably call it ourselves.